packages/utils/src/lib/timers.ts

```diff
@@ -2,39 +2,67 @@
 
 /** @public */
 export class Timers {
-	private timeouts: number[] = []
-	private intervals: number[] = []
-	private rafs: number[] = []
+	private timeouts = new Map<string, number[]>()
+	private intervals = new Map<string, number[]>()
+	private rafs = new Map<string, number[]>()
+
+	constructor() {
+		this.setTimeout = this.setTimeout.bind(this)
+		this.setInterval = this.setInterval.bind(this)
+		this.requestAnimationFrame = this.requestAnimationFrame.bind(this)
+		this.dispose = this.dispose.bind(this)
+	}
 
 	/** @public */
-	setTimeout(handler: TimerHandler, timeout?: number, ...args: any[]): number {
+	setTimeout(contextId: string, handler: TimerHandler, timeout?: number, ...args: any[]): number {
 		const id = window.setTimeout(handler, timeout, args)
-		this.timeouts.push(id)
+		const current = this.timeouts.get(contextId) ?? []
+		this.timeouts.set(contextId, [...current, id])
 		return id
 	}
 
 	/** @public */
-	setInterval(handler: TimerHandler, timeout?: number, ...args: any[]): number {
+	setInterval(contextId: string, handler: TimerHandler, timeout?: number, ...args: any[]): number {
 		const id = window.setInterval(handler, timeout, args)
-		this.intervals.push(id)
+		const current = this.timeouts.get(contextId) ?? []
+		this.intervals.set(contextId, [...current, id])
 		return id
 	}
 
 	/** @public */
-	requestAnimationFrame(callback: FrameRequestCallback): number {
+	requestAnimationFrame(contextId: string, callback: FrameRequestCallback): number {
 		const id = window.requestAnimationFrame(callback)
-		this.rafs.push(id)
+		const current = this.timeouts.get(contextId) ?? []
```

Comment:
```suggestion
		const current = this.rafs.get(contextId) ?? []
```

Reply:
this was an oversight but ironically iirc all the cleanTimeout / cancelIntervals are interchangeable